pkg/export/export.go

```diff
@@ -16,16 +31,41 @@ func pipelineExportRemoveExtraFields(input string) string {
 	return m1.ReplaceAllString(input, "\n")
 }
 
-// PipelineToYaml convert a pipeline to yaml removing extraneous fields generated by k8/tekton controllers so we can
+// TektonResourceToYaml convert a pipeline to yaml removing extraneous fields generated by k8/tekton controllers so we can
 // easily reimport theme
-func PipelineToYaml(p *v1beta1.Pipeline) (string, error) {
-	p.ObjectMeta.ManagedFields = nil
-	p.ObjectMeta.ResourceVersion = ""
-	p.ObjectMeta.UID = ""
-	p.ObjectMeta.Generation = 0
-	p.ObjectMeta.Namespace = ""
-	p.ObjectMeta.CreationTimestamp = metav1.Time{}
-
-	data, err := yaml.Marshal(p)
-	return pipelineExportRemoveExtraFields(string(data)), err
+func TektonResourceToYaml(p interface{}) (string, error) {
+	// I gave up trying to do a generic function....
```

Comment:
We can optimize it in upcoming iterations.
/lgtm

Reply:
i am not sure how to optimize tho 😅 

if someone has any ideas how to streamline it i'll be happy to take it.